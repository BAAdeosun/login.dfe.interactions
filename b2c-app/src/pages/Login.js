import React from 'react';
import components from '../components';

import { getB2CLink } from '../helpers/urls';

export default function Login() {
    return (
        
        <div id="login">
            <div className="govuk-width-container">
                <components.Breadcrumbs />

                <div id="pageLevelErrorContainer"></div>

                <div className="govuk-width-container ">
                    <main className="govuk-main-wrapper " id="main-content" role="main">
                        <div className="govuk-grid-row">
                            <div className="govuk-grid-column-one-half">
                                <components.PageTitle size='l' title='Sign in'/>
                                <components.B2C />
                            </div>
                            <div className="govuk-grid-column-one-half">
                                <components.PageTitle size='l' title='Create an account'/>
                                <p className="govuk-body">
                                    <a href={getB2CLink('signup')} className="govuk-link">Creating an account</a>
                                    &nbsp;allows you to access your adviser created action plans and save your:
                                </p>
                                <ul className="govuk-list govuk-list--bullet">
                                    <li>course searches</li>
                                    <li>Skills health check reports</li>
                                </ul>
                            </div>
                        </div>
                    </main>
                </div>
                
            </div>
        </div>
    )
}
