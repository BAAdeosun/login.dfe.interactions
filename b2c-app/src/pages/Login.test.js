
import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { JSDOM } from "jsdom";

import Login from './Login';

it('renders without crashing', () => {
    shallow(<Login />);
});

it('renders correctly', () => {
    const tree = renderer
        .create(<Login />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

describe('when submitting the form', () => {

    describe('when all details have not been set', () => {

        it('sets showErrors to true and does not call submit function', () => {
            const wrapper = mount(<Login />);
            const form = wrapper.find('#customForm');
            const setDataAndSubmitSpy = jest.spyOn(wrapper.instance(), 'setDataAndSubmit');

            form.simulate('submit');

            expect(wrapper.state().showErrors).toBe(true);
            expect(setDataAndSubmitSpy).not.toHaveBeenCalled();

        });

    });

    describe('when all details have been set', () => {

        let wrapper;
        let form;
        let setDataAndSubmitSpy;
        let b2cSubmitSpy;

        beforeEach(() => {
            //mount page with values in state for all required fields
            wrapper = mount(<Login />);

            wrapper.setState({
                email: 'a',
                password: 'a'
            });

            //get form element to simulate submit in tests
            form = wrapper.find('#customForm');

            //spy on the page submit button
            setDataAndSubmitSpy = jest.spyOn(wrapper.instance(), 'setDataAndSubmit');

            //create some DOM elements to simulate the content added by B2C
            const dom = new JSDOM();
            global.document = dom.window.document;

            //add the B2C submit button at least and spy on its click event
            let b2cSubmitElem = global.document.createElement('button');
            b2cSubmitElem.id = 'next';
            global.document.body.appendChild(b2cSubmitElem);

            //spy on b2c submit button click
            b2cSubmitSpy = jest.spyOn(global.document.getElementById('next'), 'click');
        });

        describe('when not all B2C DOM elements are present', () => {

            it('sets showErrors to false and calls submit function but does not call B2C submit', () => {

                form.simulate('submit');

                expect(setDataAndSubmitSpy).toHaveBeenCalled();
                expect(wrapper.state().showErrors).toBe(false);
                expect(b2cSubmitSpy).not.toHaveBeenCalled();
            });
        });

        describe('when all B2C DOM elements are present', () => {

            it('sets showErrors to false and calls submit function and then the B2C submit button click', () => {

                //add B2C content to the DOM
                let b2cElem = global.document.createElement('div');
                b2cElem.innerHTML = `
                <span id='email'></span>
                <span id='password'></span>
            `;
                global.document.body.appendChild(b2cElem);

                form.simulate('submit');

                expect(setDataAndSubmitSpy).toHaveBeenCalled();
                expect(wrapper.state().showErrors).toBe(false);
                expect(b2cSubmitSpy).toHaveBeenCalled();
            });

        });

    });

});
