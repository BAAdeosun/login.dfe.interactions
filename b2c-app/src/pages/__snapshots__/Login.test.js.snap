// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="login"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <div
        className="pageLevelErrorContainer hide"
      >
        <div
          aria-labelledby="error-summary-title"
          className="govuk-error-summary"
          data-module="govuk-error-summary"
          role="alert"
          tabIndex="-1"
        >
          <h2
            className="govuk-error-summary__title"
            id="error-summary-title"
          >
            There is a problem
          </h2>
          <div
            className="govuk-error-summary__body"
          >
            <ul
              className="govuk-list govuk-error-summary__list"
              id="errorSummaryItems"
            />
          </div>
        </div>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-one-half"
          >
            <h1
              className="govuk-heading-l"
            >
              Sign in
            </h1>
            <form
              id="customForm"
              noValidate={true}
              onSubmit={[Function]}
            >
              <div>
                <div
                  className="govuk-form-group "
                >
                  <label
                    className="govuk-label"
                    htmlFor="emailCustom"
                  >
                    Email address
                  </label>
                  <input
                    autoComplete="email"
                    className="govuk-input govuk-!-width-one-half"
                    id="emailCustom"
                    name="email"
                    noValidate={true}
                    onChange={[Function]}
                    spellCheck="false"
                    type="email"
                  />
                </div>
                <div
                  className="govuk-form-group "
                >
                  <label
                    className="govuk-label"
                    htmlFor="passwordCustom"
                  >
                    Password
                  </label>
                  <input
                    className="govuk-input govuk-!-width-one-half"
                    id="passwordCustom"
                    name="password"
                    noValidate={true}
                    onChange={[Function]}
                    type="password"
                  />
                </div>
                <p
                  className="govuk-body"
                >
                  <a
                    className="govuk-link"
                    href={null}
                  >
                    I cannot access my account
                  </a>
                </p>
              </div>
              <button
                className="govuk-button"
                id="preSubmit"
                type="submit"
              >
                Sign in
              </button>
            </form>
          </div>
          <div
            className="govuk-grid-column-one-half"
          >
            <h1
              className="govuk-heading-l"
            >
              Create an account
            </h1>
            <p
              className="govuk-body"
            >
              <a
                className="govuk-link"
                href={null}
              >
                Creating an account
              </a>
              Â allows you to access and save your skills health check reports.
            </p>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
